File: llm_generated_code/Gemini/CPP/Level1/Q22.cpp
#include <iostream>
#include <cmath>
#include<bits/stdc++.h>
using namespace std;

int main() {
    int a, b, c, d;
    cin >> a >> b >> c >> d;

    int gcd = __gcd(a, b);
    a /= gcd;
    b /= gcd;

    gcd = __gcd(c, d);
    c /= gcd;
    d /= gcd;

    int lcm = (a * b * c * d) / __gcd(a * d, b * c);
    int x = lcm / (a * d);
    int y = lcm / (b * c);

    cout << max(x - 1, y - 1);

    return 0;
}

File: llm_generated_code/Gemini/Python/Level1/Q22.py
from fractions import Fraction

def min_operations(a, b, c, d):
    frac1 = Fraction(a, b)
    frac2 = Fraction(c, d)
    gcd = frac1.numerator * frac2.denominator
    lcm = frac1.denominator * frac2.numerator
    return max(gcd // frac1.numerator - 1, gcd // frac2.numerator - 1,
               lcm // frac1.denominator - 1, lcm // frac2.denominator - 1)

def main():
    a, b, c, d = map(int, input().split())
    print(min_operations(a, b, c, d))

if __name__ == "__main__":
  main()
